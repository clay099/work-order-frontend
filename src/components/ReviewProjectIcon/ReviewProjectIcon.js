import React from "react";
import { useSelector } from "react-redux";
import { TableCell } from "@material-ui/core/";
import { NewReleases } from "@material-ui/icons";

const ReviewProjectIcon = ({ id }) => {
	const { bid, userType } = useSelector((st) => ({
		bid: st.bids[id],
		userType: st.login.user_type,
	}));
	return (
		<div>
			{userType === "user" && bid ? (
				<TableCell>
					<NewReleases></NewReleases>
				</TableCell>
			) : null}{" "}
		</div>
	);
};

export default ReviewProjectIcon;
