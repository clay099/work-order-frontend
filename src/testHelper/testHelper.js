// testHelper.js
import React from "react";
import { render as rtlRender } from "@testing-library/react";
import { createStore } from "redux";
import { Provider } from "react-redux";
// Import your own reducer
import reducer from "../reducers/rootReducer";

function render(
	ui,
	{
		initialState,
		store = createStore(reducer, initialState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>{children}</Provider>;
	}
	return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

const tokenState = { token: "token" };
const userTypeUserState = { userType: "user" };
const userTypeTradesmenState = { userType: "user" };
const idState = { id: 1234 };
const userDetailsState = { userDetails: "userDetails" };
const handleSubmitState = { handleSubmit: jest.fn() };
const handleChangeState = { handleChange: jest.fn() };
const disabledState = { disabled: false };
const disabledEmailState = { disabledEmail: false };
const buttonTextState = { buttonText: "buttonText" };
const formDataState = {
	formData: {
		id: 1,
		firstName: "firstName",
		lastName: "lastName",
		email: "email",
		password: "password",
		newPassword: "newPassword",
		confirmNewPassword: "confirmNewPassword",
	},
};

// re-export everything
export * from "@testing-library/react";
// override render method
export {
	render,
	tokenState,
	userTypeUserState,
	userTypeTradesmenState,
	idState,
	userDetailsState,
	handleSubmitState,
	handleChangeState,
	disabledState,
	disabledEmailState,
	buttonTextState,
	formDataState,
};
