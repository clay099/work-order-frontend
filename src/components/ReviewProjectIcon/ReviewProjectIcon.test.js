import React from "react";
import ReviewProjectIcon from "./ReviewProjectIcon";
import { render } from "../../testHelper/testHelper";
console.error = jest.fn();

const params = {
	id: 1,
};
const initialState = {
	projects: { projectList: [{ id: 1, status: "auction" }] },
	bids: { 1: 1 },
	login: { user_type: "user" },
	review: { 1: 1 },
};

describe("ReviewProjectIcon Component", () => {
	// smoke test
	test("renders without crashing", () => {
		render(<ReviewProjectIcon {...params} />, {
			initialState,
		});
	});

	// snapshot test
	test("matches snapshot", () => {
		const { asFragment } = render(<ReviewProjectIcon {...params} />, {
			initialState,
		});
		expect(asFragment).toMatchSnapshot();
	});
});
