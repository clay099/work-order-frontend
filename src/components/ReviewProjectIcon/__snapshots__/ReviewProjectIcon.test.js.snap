// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewProjectIcon Component matches snapshot 1`] = `[Function]`;
